### PR TITLE
Home task 3

### PR DESCRIPTION
**Pull request by LeraShagoyko**
Only task 4 was done in Lecture3 module. Other tasks were done in Module2

### CODE REVIEW

File: Lecture3/task2,3/files/Restaurant.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Greek Salad
```
Comment (acherkas) [inline Lecture3/task2,3/files/Restaurant.txt]: name of folder task2,3 incorrect (code convention)

File: Lecture3/task1/Exceptions/FilterException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package Exceptions;
+
+public class FilterException extends Exception {
+    public FilterException(String f){
```
Comment (acherkas) [inline Lecture3/task1/Exceptions/FilterException.java]: formatting

File: Lecture3/task1/Exceptions/NameException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package Exceptions;
+
+public class NameException extends Exception{
```
Comment (acherkas) [inline Lecture3/task1/Exceptions/NameException.java]: formatting

File: Lecture3/task1/Application.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            veganMyMySalad.addVegetable(1, new Onion());
+
+            fileWriter(greekMyMySalad.toString());
+            fileWriter(veganMyMySalad.toString());
+            fileWriter("Filtered by calories " + veganMyMySalad.getFilteredVegetablesByCalories(4, 15));
+
+            Collections.sort(greekMyMySalad.getVegetables(), new VegetablesCaloriesComparator());
+            fileWriter("Sorted by calories " + greekMyMySalad.getVegetables());
+            fileReader("Lecture3/task2,3/files/dataForFileWriter");
+        } catch (NameException nameException) {
```
Comment (acherkas) [inline Lecture3/task1/Application.java]: Java8 allows to use multiply exceptions (please read it)
 } catch (NameException | AmountException | FilterException exception) {
            System.out.println(exception.getMessage());
        }

File: Lecture3/task1/Application.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        } catch (AmountException amountException) {
+            System.out.println(amountException.getMessage());
+        } catch (FilterException filterException) {
+            System.out.println(filterException.getMessage());
+        }
+
+    }
+
+    private static List<String> fileReader(String pathToFile) {
+        List<String> listOfStrings = new ArrayList<>();
```
Comment (acherkas) [inline Lecture3/task1/Application.java]: incorrect work of file reader
read examples
https://www.journaldev.com/709/java-read-file-line-by-line

File: Lecture3/task1/Application.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                listOfStrings.add(line);
+            }
+        } catch (IOException ex) {
+
+            System.out.println(ex.getMessage());
+        }
+        return listOfStrings;
+    }
+
+    private static void fileWriter(String inputMessage) {
```
Comment (acherkas) [inline Lecture3/task1/Application.java]: extract all static methods to separate class (Helper)

Comment (acherkas) [inline Lecture3/task1/Application.java]: rename to writeToFile

File: Lecture3/task1/Application.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+            System.out.println(nameException.getMessage());
+        } catch (AmountException amountException) {
+            System.out.println(amountException.getMessage());
+        } catch (FilterException filterException) {
+            System.out.println(filterException.getMessage());
+        }
+
+    }
+
+    private static List<String> fileReader(String pathToFile) {
```
Comment (acherkas) [inline Lecture3/task1/Application.java]: rename to readFromFile

File: Lecture3/task1/Salad.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class Salad {
+    private String name;
+    private ArrayList<Vegetable> vegetables = new ArrayList<>();
+    private int totalCalories;
+
+    public Salad(String name) throws NameException{
+        if (name == "") {
```
Comment (acherkas) [inline Lecture3/task1/Salad.java]: check how we should compare the strings

File: Lecture3/task1/Application.java (hunk 6 above)
Comment (acherkas) [inline Lecture3/task1/Application.java]: please be sure that each run of program should reset the content of file

File: Lecture3/Task4/GreekMyMySalad.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+public class GreekMyMySalad implements MySalad {
```
Comment (acherkas) [inline Lecture3/Task4/GreekMyMySalad.java]: rename to GreekSalad

File: Lecture3/Task4/CheeseMyMySalad.java [hunk 10]
```diff
@@ -0,0 +1,7 @@
+public class CheeseMyMySalad implements MySalad {
```
Comment (acherkas) [inline Lecture3/Task4/CheeseMyMySalad.java]: rename to CheeseSalad

File: Lecture3/Task4/VeganMyMySalad.java [hunk 11]
```diff
@@ -0,0 +1,6 @@
+public class VeganMyMySalad implements MySalad {
```
Comment (acherkas) [inline Lecture3/Task4/VeganMyMySalad.java]: rename to VeganSalad

File: Lecture3/task2,3/files/Restaurant.txt (hunk 1 above)
Comment (LeraShagoyko) [reply Lecture3/task2,3/files/Restaurant.txt]: renamed

File: Lecture3/task1/Exceptions/FilterException.java (hunk 2 above)
Comment (LeraShagoyko) [reply Lecture3/task1/Exceptions/FilterException.java]: done

File: Lecture3/task1/Exceptions/NameException.java (hunk 3 above)
Comment (LeraShagoyko) [reply Lecture3/task1/Exceptions/NameException.java]: done

File: Lecture3/task1/Application.java (hunk 4 above)
Comment (LeraShagoyko) [reply Lecture3/task1/Application.java]: fixed

File: Lecture3/task1/Application.java (hunk 6 above)
Comment (LeraShagoyko) [reply Lecture3/task1/Application.java]: done

Comment (LeraShagoyko) [reply Lecture3/task1/Application.java]: renamed

File: Lecture3/task1/Application.java (hunk 7 above)
Comment (LeraShagoyko) [reply Lecture3/task1/Application.java]: renamed

File: Lecture3/task1/Salad.java (hunk 8 above)
Comment (LeraShagoyko) [reply Lecture3/task1/Salad.java]: done

File: Lecture3/task1/Application.java (hunk 6 above)
Comment (LeraShagoyko) [reply Lecture3/task1/Application.java]: fixed

File: Lecture3/Task4/GreekMyMySalad.java (hunk 9 above)
Comment (LeraShagoyko) [reply Lecture3/Task4/GreekMyMySalad.java]: renamed

File: Lecture3/Task4/CheeseMyMySalad.java (hunk 10 above)
Comment (LeraShagoyko) [reply Lecture3/Task4/CheeseMyMySalad.java]: renamed

File: Lecture3/Task4/VeganMyMySalad.java (hunk 11 above)
Comment (LeraShagoyko) [reply Lecture3/Task4/VeganMyMySalad.java]: renamed

File: Lecture3/task1/Application.java (hunk 5 above)
Comment (LeraShagoyko) [reply Lecture3/task1/Application.java]: fixed

**Pull request merged**